src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedProcessor.java

```diff
@@ -60,7 +63,16 @@ public void processQuestionnaireLinked(ResponseManagementEvent questionnaireLink
     UacQidLink uacQidLink = uacQidLinkOpt.get();
     Case caze = caseOpt.get();
 
+    // Handle when linking happens after case has been receipted
+    if (caze.isReceiptReceived() && !uacQidLink.isActive()) {
```

Comment:
if the case receipt received is already set true, then why do we need to set it true again and emit a case updated event?